src/AdornedRulerPanel.h

```diff
@@ -15,6 +15,7 @@
 #include "widgets/Ruler.h" // member variable
 #include "Prefs.h"
 #include "ViewInfo.h" // for PlayRegion
+#include "PlaybackLooping.h"
```

Comment:
You should not need to include a header inside another header just to declare a shared_ptr.  A forward declaration of an incomplete type would be better.